### PR TITLE
Add ability to specify path to OIDC token

### PR DESCRIPTION
**Pull request by mpminardi**
Adds the ability to specify a path to an OIDC token on disk in place of supplying the token directly to the provider.

Related PR in go-azure-helpers: https://github.com/hashicorp/go-azure-helpers/pull/131/files

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -248,6 +248,13 @@ func azureProvider(supportLegacyTestSuite bool) *schema.Provider {
 				Description: "The OIDC ID token for use when authenticating as a Service Principal using OpenID Connect.",
 			},
 
+			"oidc_token_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_OIDC_TOKEN_PATH", ""),
+				Description: "The path to an OIDC ID token for use when authenticating as a Service Principal using OpenID Connect.",
```
Comment (katbyte) [inline internal/provider/provider.go:255]:
```suggestion
				Description: "The path to a file containing an OIDC ID token for use when authenticating as a Service Principal using OpenID Connect.",
```

File: internal/provider/provider.go [hunk 2]
```diff
@@ -248,6 +248,13 @@ func azureProvider(supportLegacyTestSuite bool) *schema.Provider {
 				Description: "The OIDC ID token for use when authenticating as a Service Principal using OpenID Connect.",
 			},
 
+			"oidc_token_path": {
```
Comment (katbyte) [inline internal/provider/provider.go:251]: would this be better as
```suggestion
			"oidc_token_file_path": {
```

File: internal/provider/provider.go [hunk 3]
```diff
@@ -248,6 +248,13 @@ func azureProvider(supportLegacyTestSuite bool) *schema.Provider {
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_OIDC_TOKEN_PATH", ""),
```
Comment (katbyte) [inline internal/provider/provider.go:254]:
```suggestion
				DefaultFunc: schema.EnvDefaultFunc("ARM_OIDC_TOKEN_FILE_PATH", ""),
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @caius - juts a couple minor comments around the schema

**Review by manicminer — APPROVED** [review]
Thanks for the contribution @mpminardi! I've gone ahead and vendored in the upstream change and merged in main. Just waiting on the checks & unit tests but this LGTM 👍

**Pull request merged**